In [2]:
# IMPORT THE LIBRARIES

In [3]:
import pandas as pd
import sqlite3

In [4]:
# CREATE DATABASE CONNECTIONS

In [5]:
conn = sqlite3.connect('ipl_data.db')
print("Database connected successfully")

Database connected successfully


In [6]:
# LOAD CSV FILES

In [7]:
matches = pd.read_csv(r"H:\project\matches.csv")
deliveries = pd.read_csv(r"H:\project\deliveries.csv")

print("CSV files Extracted successfully !")
print(f"Matches: {matches.shape}")
print(f"Deliveries: {deliveries.shape}")

CSV files Extracted successfully !
Matches: (1095, 20)
Deliveries: (260920, 17)


In [8]:
# CHECK INCONSISTENCY TEAM NAMES

In [9]:
print(pd.concat([matches['team1'], matches['team2']]).unique())

<StringArray>
['Royal Challengers Bangalore',             'Kings XI Punjab',
            'Delhi Daredevils',              'Mumbai Indians',
       'Kolkata Knight Riders',            'Rajasthan Royals',
             'Deccan Chargers',         'Chennai Super Kings',
        'Kochi Tuskers Kerala',               'Pune Warriors',
         'Sunrisers Hyderabad',               'Gujarat Lions',
     'Rising Pune Supergiants',      'Rising Pune Supergiant',
              'Delhi Capitals',                'Punjab Kings',
        'Lucknow Super Giants',              'Gujarat Titans',
 'Royal Challengers Bengaluru']
Length: 19, dtype: str


In [10]:
# STANDARDIZE TEAM NAMES

In [15]:
matches.replace({
    'Royal Challengers Bangalore' : 'Royal Challengers Bengaluru',
    'Delhi Daredevils'            : 'Delhi Daredevils',
    'Kings XI Punjab'             : 'Punjab Kings',
    'Pune Warriors'               : 'Rising Pune Supergiants',
    'Rising Pune Supergiant'      : 'Rising Pune Supergiants'
})
deliveries.replace({
    'Royal Challengers Bangalore' : 'Royal Challengers Bengaluru',
    'Delhi Daredevils'            : 'Delhi Daredevils',
    'Kings XI Punjab'             : 'Punjab Kings',
    'Pune Warriors'               : 'Rising Pune Supergiants',
    'Rising Pune Supergiant'      : 'Rising Pune Supergiants'
})

,match_id,inning,batting_team,bowling_team,over,ball,batter,bowler,non_striker,batsman_runs,extra_runs,total_runs,extras_type,is_wicket,player_dismissed,dismissal_kind,fielder
0,335982,1,Kolkata Knight Riders,Royal Challengers Bengaluru,0,1,SC Ganguly,P Kumar,BB McCullum,0,1,1,legbyes,0,NaN,NaN,NaN
1,335982,1,Kolkata Knight Riders,Royal Challengers Bengaluru,0,2,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,NaN,NaN,NaN
2,335982,1,Kolkata Knight Riders,Royal Challengers Bengaluru,0,3,BB McCullum,P Kumar,SC Ganguly,0,1,1,wides,0,NaN,NaN,NaN
3,335982,1,Kolkata Knight Riders,Royal Challengers Bengaluru,0,4,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,NaN,NaN,NaN
4,335982,1,Kolkata Knight Riders,Royal Challengers Bengaluru,0,5,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
260915,1426312,2,Kolkata Knight Riders,Sunrisers Hyderabad,9,5,SS Iyer,AK Markram,VR Iyer,1,0,1,NaN,0,NaN,NaN,NaN
260916,1426312,2,Kolkata Knight Riders,Sunrisers Hyderabad,9,6,VR Iyer,AK Markram,SS Iyer,1,0,1,NaN,0,NaN,NaN,NaN
260917,1426312,2,Kolkata Knight Riders,Sunrisers Hyderabad,10,1,VR Iyer,Shahbaz Ahmed,SS Iyer,1,0,1,NaN,0,NaN,NaN,NaN
260918,1426312,2,Kolkata Knight Riders,Sunrisers Hyderabad,10,2,SS Iyer,Shahbaz Ahmed,VR Iyer,1,0,1,NaN,0,NaN,NaN,NaN


In [16]:
# CREATE TABLES IN SQLITE DATABASE

In [17]:
matches.to_sql('matches', conn, index=False, if_exists='replace')
deliveries.to_sql('deliveries', conn, index=False, if_exists='replace')

print("Tables created: 'matches' and 'deliveries'")

Tables created: 'matches' and 'deliveries'


In [18]:
# VERIFY TABLES IN DATABASE

In [19]:
tables = pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table'", conn)
print("Tables in database:")
print(tables)

Tables in database:
         name
0     matches
1  deliveries


In [20]:
# FIRST SELECT QUERIES

In [21]:
# Query 1: First 5 rows of matches
q1 = pd.read_sql_query("SELECT * FROM matches LIMIT 5", conn)
print("Matches (first 5 rows):")
display(q1)

# Query 2: First 10 rows of deliveries
q2 = pd.read_sql_query("SELECT * FROM deliveries LIMIT 10", conn)
print("\nDeliveries (first 10 rows):")
display(q2)

# Query 3: Select specific columns
q3 = pd.read_sql_query("SELECT season, team1, team2, winner FROM matches LIMIT 5", conn)
print("\nSelected columns from matches:")
display(q3)

Matches (first 5 rows):


,id,season,city,date,match_type,player_of_match,venue,team1,team2,toss_winner,toss_decision,winner,result,result_margin,target_runs,target_overs,super_over,method,umpire1,umpire2
0,335982,2007/08,Bangalore,2008-04-18,League,BB McCullum,M Chinnaswamy Stadium,Royal Challengers Bangalore,Kolkata Knight Riders,Royal Challengers Bangalore,field,Kolkata Knight Riders,runs,140.0,223.0,20.0,N,None,Asad Rauf,RE Koertzen
1,335983,2007/08,Chandigarh,2008-04-19,League,MEK Hussey,"Punjab Cricket Association Stadium, Mohali",Kings XI Punjab,Chennai Super Kings,Chennai Super Kings,bat,Chennai Super Kings,runs,33.0,241.0,20.0,N,None,MR Benson,SL Shastri
2,335984,2007/08,Delhi,2008-04-19,League,MF Maharoof,Feroz Shah Kotla,Delhi Daredevils,Rajasthan Royals,Rajasthan Royals,bat,Delhi Daredevils,wickets,9.0,130.0,20.0,N,None,Aleem Dar,GA Pratapkumar
3,335985,2007/08,Mumbai,2008-04-20,League,MV Boucher,Wankhede Stadium,Mumbai Indians,Royal Challengers Bangalore,Mumbai Indians,bat,Royal Challengers Bangalore,wickets,5.0,166.0,20.0,N,None,SJ Davis,DJ Harper
4,335986,2007/08,Kolkata,2008-04-20,League,DJ Hussey,Eden Gardens,Kolkata Knight Riders,Deccan Chargers,Deccan Chargers,bat,Kolkata Knight Riders,wickets,5.0,111.0,20.0,N,None,BF Bowden,K Hariharan



Deliveries (first 10 rows):


,match_id,inning,batting_team,bowling_team,over,ball,batter,bowler,non_striker,batsman_runs,extra_runs,total_runs,extras_type,is_wicket,player_dismissed,dismissal_kind,fielder
0,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,1,SC Ganguly,P Kumar,BB McCullum,0,1,1,legbyes,0,None,None,None
1,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,2,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,None,None,None
2,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,3,BB McCullum,P Kumar,SC Ganguly,0,1,1,wides,0,None,None,None
3,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,4,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,None,None,None
4,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,5,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,None,None,None
5,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,6,BB McCullum,P Kumar,SC Ganguly,0,0,0,NaN,0,None,None,None
6,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,7,BB McCullum,P Kumar,SC Ganguly,0,1,1,legbyes,0,None,None,None
7,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,1,1,BB McCullum,Z Khan,SC Ganguly,0,0,0,NaN,0,None,None,None
8,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,1,2,BB McCullum,Z Khan,SC Ganguly,4,0,4,NaN,0,None,None,None
9,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,1,3,BB McCullum,Z Khan,SC Ganguly,4,0,4,NaN,0,None,None,None



Selected columns from matches:


,season,team1,team2,winner
0,2007/08,Royal Challengers Bangalore,Kolkata Knight Riders,Kolkata Knight Riders
1,2007/08,Kings XI Punjab,Chennai Super Kings,Chennai Super Kings
2,2007/08,Delhi Daredevils,Rajasthan Royals,Delhi Daredevils
3,2007/08,Mumbai Indians,Royal Challengers Bangalore,Royal Challengers Bangalore
4,2007/08,Kolkata Knight Riders,Deccan Chargers,Kolkata Knight Riders


In [22]:
# CLOSE DATABASE CONNECTION

In [23]:
conn.close()
print("Connection closed.")

Connection closed.


## Day 1 Summary: SQLite Setup

### Completed
- Created SQLite database (`ipl_data.db`)
- Loaded `matches.csv` (1,095 rows, 20 columns) into `matches` table
- Loaded `deliveries.csv` (260,920 rows, 17 columns) into `deliveries` table
- Standardized RCB team name (Bangalore → Bengaluru) across all columns
- Verified table creation
- Ran first SELECT queries to explore both tables

### Database Schema

**matches** — IPL match summaries (2008–2024)
- Columns: id, season, city, date, team1, team2, toss_winner, toss_decision, winner, result, venue, player_of_match, etc.

**deliveries** — Ball-by-ball data
- Columns: match_id, inning, batting_team, bowling_team, over, ball, batter, bowler, batsman_runs, is_wicket, etc.

### Next
Day 2 — Filtering with WHERE: Extract all RCB matches and count them